### PR TITLE
#1518 - Data Type Periods

### PR DESCRIPTION
**Pull request by JohnPTobe**
##### Checklist

- [X] `manage.py test` passes
- [X] tests are included
- [X] migrations are included
- [X] documentation is changed or added

### Affected app(s)
- ingest
- storage

### Description of change
Removed the data tag pattern check for data type tags and changed the field in the database from a CharField to an array of CharFields so we don't care what characters are used for tags and we don't have to do any splitting on a comma separated string.

### CODE REVIEW

**Comment by JohnPTobe** [comment]
Ready for review

File: scale/storage/serializers.py [hunk 1]
```diff
@@ -112,7 +91,7 @@ class ScaleFileBaseSerializerV5(ModelIdSerializer):
     media_type = serializers.CharField()
     file_type = serializers.CharField()
     file_size = serializers.IntegerField()  # TODO: BigIntegerField?
-    data_type = DataTypeField()
+    data_type = serializers.ListField(child=serializers.CharField(), source='data_type_tags')
```
Comment (emimaesmith) [inline scale/storage/serializers.py]: Is there a reason this was left as `data_type` instead of `data_type_tags`?

Comment (JohnPTobe) [reply scale/storage/serializers.py]: It's pointless as v5 is going away but I kept it the field name the same for the v5 serializer out of habit.

Comment (JohnPTobe) [reply scale/storage/serializers.py]: The v6 field has the name change.

**Pull request merged**